arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+                var totalCustomers = 0
+                var totalLoad = 0
+                results.forEach { result in
+                    switch result {
+                    case let elementResult as AGSUtilityElementTraceResult:
+                        // Get the unique customers count.
+                        totalCustomers = Set(elementResult.elements.map(\.objectID)).count
+                    case let functionResult as AGSUtilityFunctionTraceResult:
+                        // Get the total load with a function output.
+                        totalLoad = functionResult.functionOutputs.first?.result as? Int ?? 0
+                    default:
+                        break
+                    }
+                }
+                self.summaries[phase] = PhaseSummary(totalCustomers: totalCustomers, totalLoad: totalLoad)
+            }
+        }
+        // Reload the load report table when trace completes.
+        traceGroup.notify(queue: .main) { [weak self] in
+            SVProgressHUD.dismiss()
+            self?.tableView.reloadSections([Section.included.rawValue], with: .automatic)
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["CreateLoadReportViewController"]
+        // Load the utility network and initialize properties.
+        loadUtilityNetwork()
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension CreateLoadReportViewController: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        Section.allCases[section].titleForHeader
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        Section.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section.allCases[section] {
+        case .included:
+            return includedPhases.count
+        case .excluded:
+            return excludedPhases.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        let phase: AGSCodedValue
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.included.cellIdentifier, for: indexPath)
+            phase = includedPhases[indexPath.row]
+            cell.textLabel?.text = "Phase: " + phase.name
+            if let summary = summaries[phase] {
+                cell.detailTextLabel?.text = "C: " + numberFormatter.string(from: NSNumber(value: summary.totalCustomers))! + "    L: " + numberFormatter.string(from: NSNumber(value: summary.totalLoad))!
```

Comment:
Not very happy with the strings here, but I don't know where to go 🤔 
Also I haven't updated the screenshot, in case further UI updates to make.

Reply:
This is still string concatenation, which is not a good idea in a localized environment. Again, not a problem for this project because it isn't localized, but I'd hate for someone to copy what we've done in a localized app. The way to go is to use a _format string_. If you do a search for `String(format:`, you'll find plenty of examples in our samples.